File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, (len(nums) - 1)
        while l <= r:
            m = (l + r) // 2
            if (nums[l] <= nums[m]) and (nums[m] <= nums[r]):
                return nums[l]
            elif (nums[l] >= nums[m]) and (nums[m] >= nums[r]):
                return nums[r]
            elif (nums[l] >= nums[m]) and (nums[m] <= nums[r]):
                r = m
            else:
                l = m
        return None

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        if (nums.empty()){
            return 0;
        }

        int curRes = 0;
        int res = INT_MAX;
        int l = 0;
        for (int r = 0; r < nums.size(); r++){
            curRes += nums[r];
            while (curRes >= target){
                res = min(res, r - l + 1);
                curRes -= nums[l];
                l++;
            }
        }
        if (res == INT_MAX){
            return 0;
        }
        return res;
    }
};